LunaDll/LuaMain/LuaProxyFFI.cpp

```diff
@@ -568,6 +569,74 @@ typedef struct ExtendedBlockFields_\
         std::wstring wpath = Str2WStr(path);
         return gCachedFileMetadata.exists(wpath);
     }
+
+    FFI_EXPORT(void) LunaLuaSetWindowTitle(const char* newName)
+    {
+        // Add "Software Rendered" if using it, like the normal window name code does
+        std::string fullName = newName;
+        if (gStartupSettings.softwareGL)
+        {
+            fullName += " (Software Renderer)";
+        }
+
+        // Main bit for setting window name
+        SetWindowLongW(gMainWindowHwnd,GWL_WNDPROC,GetWindowLongA(gMainWindowHwnd,GWL_WNDPROC));
+        SetWindowTextW(gMainWindowHwnd,Str2WStr(fullName).c_str());
+
+        // ... But we also set these three VB-related ones, since otherwise
+        // they seem to cause a weird issue where it can sometimes (and
+        // only sometimes) get reset after refocusing the window.
+        GM_GAMETITLE_1 = fullName;
+        GM_GAMETITLE_2 = fullName;
+        GM_GAMETITLE_3 = fullName;
```

Comment:
Try to do next:
```cpp
std::wstring fullNameW = Str2WStr(fullName);
SetWindowTextW(gMainWindowHwnd, fullNameW.c_str());

GM_GAMETITLE_1 = fullNameW;
GM_GAMETITLE_2 = fullNameW;
GM_GAMETITLE_3 = fullNameW;
```

Reply:
Unfortunately, there still seems to be no difference. The GAMETITLE addresses only affect some weird and specific edge cases, though it doesn't change those either.

Reply:
There's no need to convert to std::wstring before assigning to the GM_GAMETITLE_* things. It's slightly more efficient to do so, as to only do the conversion once, but assigning a std::string to VBStrPtr as was previously done automatically does a Str2WStr behind the scenes. There's no reason to have expect that to have made any difference.